File: ios/MapTalk/Core/RelativeTime.swift
import Foundation

/// Short, chat style timestamps: "now", "4m", "3h", "2d".
func relativeTime(_ date: Date?, now: Date = Date()) -> String {
    guard let date else { return "sending\u{2026}" }
    let seconds = max(0, now.timeIntervalSince(date))
    let minutes = Int(seconds / 60)
    switch minutes {
    case ..<1: return "now"
    case ..<60: return "\(minutes)m"
    default: break
    }
    let hours = minutes / 60
    if hours < 24 { return "\(hours)h" }
    let days = hours / 24
    return days < 7 ? "\(days)d" : "\(days / 7)w"
}

File: android/app/src/main/java/app/maptalk/ui/RelativeTime.kt
package app.maptalk.ui

import java.time.Duration
import java.time.Instant

/** Short, chat style timestamps: "now", "4m", "3h", "2d". */
fun relativeTime(instant: Instant?, now: Instant = Instant.now()): String {
    if (instant == null) return "sending\u2026"
    val elapsed = Duration.between(instant, now)
    val minutes = elapsed.toMinutes()
    return when {
        minutes < 1 -> "now"
        minutes < 60 -> "${minutes}m"
        elapsed.toHours() < 24 -> "${elapsed.toHours()}h"
        elapsed.toDays() < 7 -> "${elapsed.toDays()}d"
        else -> "${elapsed.toDays() / 7}w"
    }
}
